app/src/main/java/edu/uc/forbesne/shoppingsidekick/ui/main/MainViewModel.kt

```diff
@@ -1,53 +1,206 @@
 package edu.uc.forbesne.shoppingsidekick.ui.main
+// code is based on professor's github - https://github.com/discospiff/MyPlantDiaryQ
 
 import android.util.Log
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import edu.uc.forbesne.shoppingsidekick.dto.*
 import com.google.firebase.firestore.FirebaseFirestore
 import com.google.firebase.firestore.FirebaseFirestoreSettings
+import edu.uc.forbesne.shoppingsidekick.dto.Cart
 import edu.uc.forbesne.shoppingsidekick.dto.CartItem
 import edu.uc.forbesne.shoppingsidekick.dto.Product
+import edu.uc.forbesne.shoppingsidekick.dto.SearchItem
 import edu.uc.forbesne.shoppingsidekick.service.ProductService
 
-//I used the professor's code from his github - https://github.com/discospiff/MyPlantDiaryQ
-// as the base for this code
 
 /**
- *  Gets data from server, makes adjustments, provides live data to activities
- *
+ *  Gets data from Firebase and APIs, makes adjustments, provides live data to activities
  */
 class MainViewModel : ViewModel() {
-
-    var products: MutableLiveData<ArrayList<Product>> = MutableLiveData<ArrayList<Product>>()
-    var productService: ProductService = ProductService()
     private lateinit var firestore : FirebaseFirestore
+    var productService: ProductService = ProductService()
+    var products: MutableLiveData<ArrayList<Product>> = MutableLiveData<ArrayList<Product>>()
+    var productsFromShop1: MutableLiveData<ArrayList<Product>> = MutableLiveData<ArrayList<Product>>()
+    var productsFromShop2: MutableLiveData<ArrayList<Product>> = MutableLiveData<ArrayList<Product>>()
+    var productsFromShop3: MutableLiveData<ArrayList<Product>> = MutableLiveData<ArrayList<Product>>()
+    var cart: Cart = Cart(ArrayList<CartItem>())
+
+    var searchItemList: ArrayList<SearchItem> = ArrayList<SearchItem>()
+
+    // A list of object{shopName, ProductUPC, shopsPrice}
+    var initialProductPriceList :ProductPriceList = ProductPriceList(3)
+    // A map/table containing all key value pairs where key = productUPC, value = list of objects {shopName, ProductUPC,shopPrice}
+    var productPricesByShopMap: HashMap<String,ProductPriceList> = HashMap<String, ProductPriceList>()
 
     init {
-        fetchAllProducts()
+        createFirebaseInstance()
```

Comment:
I moved the creation of the firebase instance into a function as I think it could be helpful when we try to mocking in the tests, as functions can be mocked.